lib/puppetserver/ca/action/import.rb

```diff
@@ -48,41 +51,71 @@ def run(input)
 
           settings_overrides = {}
           settings_overrides[:certname] = input['certname'] unless input['certname'].empty?
+          settings_overrides[:dns_alt_names] = input['subject-alt-names'] unless input['subject-alt-names'].empty?
+
           puppet = Config::Puppet.new(config_path)
           puppet.load(settings_overrides)
           return 1 if CliParsing.handle_errors(@logger, puppet.errors)
 
-          target_locations = [puppet.settings[:cacert],
-                              puppet.settings[:cakey],
-                              puppet.settings[:cacrl],
-                              puppet.settings[:serial],
-                              puppet.settings[:cert_inventory]]
-          errors = FileSystem.check_for_existing_files(target_locations)
+          # Load most secure signing digest we can for cers/crl/csr signing.
+          signer = SigningDigest.new
+          return 1 if CliParsing.handle_errors(@logger, signer.errors)
+
+          errors = import(loader, puppet.settings, signer.digest)
+          return 1 if CliParsing.handle_errors(@logger, errors)
+
+          @logger.inform "Import succeeded. Find your files in #{puppet.settings[:cadir]}"
+          return 0
+        end
+
+        def import(loader, settings, signing_digest)
+          ca = Puppetserver::Ca::LocalCertificateAuthority.new(signing_digest, settings)
+          master_key, master_cert = ca.create_master_cert(loader.key, loader.certs.first)
+
+          FileSystem.ensure_dirs([settings[:cadir],
+                                  settings[:certdir],
+                                  settings[:privatekeydir],
+                                  settings[:publickeydir],
+                                  settings[:signeddir]])
+
+          public_files = [
+            [settings[:cacert], loader.certs],
+            [settings[:cacrl], loader.crls],
+            [settings[:localcacert], loader.certs],
+            [settings[:hostcrl], loader.crls],
+            [settings[:hostpubkey], master_key.public_key],
+            [settings[:hostcert], master_cert],
+            [settings[:cert_inventory], ca.inventory_entry(master_cert)],
+            [settings[:serial], "0x0002"],
+            [File.join(settings[:signeddir], "#{settings[:certname]}.pem"), master_cert]
+          ]
```

Comment:
So added in here localcacert and hostcrl (same as cacert and cacrl) along with the Master cert stuff we need (hostcert, hostpubkey) Also, instead of ensuring the serial and cert inventory, I just create them since we have custom stuff we want in there 
(instead of
```
 FileSystem.ensure_file(puppet.settings[:serial], "001", 0640)
 FileSystem.ensure_file(puppet.settings[:cert_inventory], "", 0640)
```
)